src/Chainweb/Pact/PactService.hs

```diff
@@ -517,24 +517,18 @@ type BuyGasValidation = T3 ChainwebTransaction Uniqueness AbleToBuyGas
 --
 attemptBuyGas
     :: PayloadCas cas
-    => Checkpointer
-    -> Miner
+    => Miner
     -> Maybe BlockHash
+    -> PactDbEnv'
     -> Vector (T2 ChainwebTransaction Uniqueness)
     -> PactServiceM cas (Vector BuyGasValidation)
-attemptBuyGas cp miner nonGenesisParentHash txs = withDiscardedBatch $ do
-    mbLatestBlock <- liftIO $ _cpGetLatestBlock cp
-    (bh, bhash) <- case mbLatestBlock of
-        Nothing -> throwM NoBlockValidatedYet
-        Just !p -> return p
-    let target = Just (succ bh, bhash)
-    withCheckpointer target "attemptBuyGas" $ \(PactDbEnv' dbEnv) -> do
+attemptBuyGas miner nonGenesisParentHash (PactDbEnv' dbEnv) txs = do
```

Comment:
:+1: this was a good catch that we were unintentionally recursing on the savepoints here. _WHY_ sqlite lets you do this instead of throwing an error is beyond me